docs/reference/modules/coordination.asciidoc

```diff
@@ -0,0 +1,497 @@
+obtain confirmation that they are no longer in the voting configuration.  In
+the example described above, shrinking a seven-master-node cluster down to only
+have three master nodes, you could retire four of the nodes and then shut them
+down simultaneously. 
+
+A node (or nodes) can be brought back out of retirement using the `unretire`
+API:
+
+[source,js]
+--------------------------------------------------
+POST /_nodes/node_name/unretire
+--------------------------------------------------
+// CONSOLE
+
+The node (or nodes) to reinstate are specified using <<cluster-nodes,node
+filters>> in place of `node_name` here. After being brought back out of
+retirement they might or might not immediately be added to the voting
+configuration.
+
+The current set of retired nodes is stored in the cluster state and can be
+inspected as follows:
+
+[source,js]
+--------------------------------------------------
+GET /_cluster/state?filter_path=TODO
+--------------------------------------------------
+// CONSOLE
+
+This set is limited in size by the following setting:
+
+`cluster.max_retired_nodes`::
+
+    Sets a limits on the number of retired nodes at any one time. Defaults to
+    `10`.
+
+Because there can only be a limited number of retired nodes at once, once a
+retired node has been destroyed its entry should be removed from the set of
+retired nodes using the unretire API.
+
+[float]
+=== Cluster bootstrapping
+
+There is a risk when starting up a brand-new cluster is that you accidentally
+form two separate clusters instead of one. This could lead to data loss: you
+might start using both clusters before noticing that anything had gone wrong,
+and it will then be impossible to merge them together later.
+
+NOTE: To illustrate how this could happen, imagine starting up a three-node
+cluster in which each node knows that it is going to be part of a three-node
+cluster. A majority of three nodes is two, so normally the first two nodes to
+discover each other will form a cluster and the third node will join them a
+short time later. However, imagine that four nodes were accidentally started
+instead of three: in this case there are enough nodes to form two separate
+clusters. Of course if each node is started manually then it's unlikely that
+too many nodes are started, but it's certainly possible to get into this
+situation if using a more automated orchestrator, particularly if a network
+partition happens at the wrong time.
+
+We avoid this by requiring a separate _cluster bootstrapping_ process to take
+place on every brand-new cluster. This is only required the very first time the
+whole cluster starts up: new nodes joining an established cluster can safely
+obtain all the information they need from the elected master, and nodes that
+have previously been part of a cluster will have stored to disk all the
+information required when restarting.
+
+A cluster can be bootstrapped by sending a _bootstrap warrant_ to any of its
```

Comment:
this will scare non-native speakers ;)
How about just calling it the initial voting configuration?

Reply:
Renamed to `bootstrap configuration`.